File: net/Functions/Startup.fs


namespace Functions
open System
open Owin
open System.Web.Http;

type public Startup() = 
    member this.Configuration (appBuilder : IAppBuilder) = 
        // Configure Web API for self-host. 
        let config = new HttpConfiguration()
        config.MapHttpAttributeRoutes()
        appBuilder.UseWebApi(config) |> ignore

File: net/ClassesAndInterfaces/Startup.cs
using Owin;
using System.Web.Http;
using Autofac;
using Autofac.Integration.WebApi;
using System.Reflection;

namespace ClassesAndInterfaces
{
    public class Startup
    {
        // This code configures Web API. The Startup class is specified as a type
        // parameter in the WebApp.Start method.
        public void Configuration(IAppBuilder appBuilder)
        { 
            var builder = new ContainerBuilder();

            // Configure Web API for self-host. 
            HttpConfiguration config = new HttpConfiguration(); 
            config.MapHttpAttributeRoutes(); 

            builder.RegisterApiControllers(Assembly.GetExecutingAssembly());
            builder.RegisterType<WeaponMemoryRepository>().As<IWeaponRepository>();
            var container = builder.Build();
            config.DependencyResolver = new AutofacWebApiDependencyResolver(container);

            appBuilder.UseWebApi(config); 
        }
    }
}

